sdk/cosmos/azure-cosmos/azure/cosmos/aio/_cosmos_client_connection_async.py

```diff
@@ -232,10 +224,54 @@ def _ReadEndpoint(self):
         return self._global_endpoint_manager.get_read_endpoint()
 
     async def _setup(self):
+        # Save the choice that was made (either None or some value) and branch to set or get the consistency
+        user_defined_consistency = self.default_headers[http_constants.HttpHeaders.ConsistencyLevel]
+        # Set header setting to lowest latency consistency level for the first request to _GetDatabaseAccount
+        if user_defined_consistency is None:
+            self.default_headers[http_constants.HttpHeaders.ConsistencyLevel] = documents.ConsistencyLevel.Eventual
+
+        if user_defined_consistency == documents.ConsistencyLevel.Session:
+            # create a Session if the user wants Session consistency
+            self.session = _session.Session(self.url_connection)
+        else:
+            self.session = None  # type: ignore
+
+        database_account = None
         if 'database_account' not in self._setup_kwargs:
-            self._setup_kwargs['database_account'] = await self._global_endpoint_manager._GetDatabaseAccount(
+            database_account = await self._global_endpoint_manager._GetDatabaseAccount(
                 **self._setup_kwargs)
+            self._setup_kwargs['database_account'] = database_account
             await self._global_endpoint_manager.force_refresh(self._setup_kwargs['database_account'])
+        else:
+            database_account = self._setup_kwargs.get('database_account')
+
+        # Use database_account if no consistency passed in to verify consistency level to be used
+        if user_defined_consistency is None:
+            self._set_account_consistency_level(database_account)
+
+    def _set_account_consistency_level(
+            self,
+            database_account: ClassType,
+    ) -> None:
+        """Checks if consistency level param was passed in by user and sets it to that value or to the account default.
+        :param database_account: The database account to be used to check consistency levels
+        :type database_account: ~azure.cosmos.documents.DatabaseAccount
+        :rtype: None
+        """
+        # Set to default level present in account
+        user_consistency_policy = database_account.ConsistencyPolicy
+        consistency_level = user_consistency_policy.get(constants._Constants.DefaultConsistencyLevel)
+
+        self.default_headers[http_constants.HttpHeaders.ConsistencyLevel] = consistency_level
```

Comment:
Should it be setting the consistency level header if it using the default account consistency level?